src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,188 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
-  return (
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+import { PersonLink } from './PersonLink';
+import { useMemo } from 'react';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+  const gender = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const sortedOptions = ['Name', 'Sex', 'Born', 'Died'];
+
+  const filteredPeople = useMemo(() => {
+    let filtered = [...people];
+
+    if (query) {
+      const normalizedQuery = query.toLocaleLowerCase();
```

Comment:
Instead of using `toLocaleLowerCase()` method multiple times on the same string, you could store the result in a variable and use it in your filter function. This would improve the performance of your code.